x-pack/plugins/encrypted_saved_objects/server/config.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginInitializerContext } from 'src/core/server';
+
+export type ConfigType = ReturnType<typeof createConfig$> extends Observable<infer P>
+  ? P
+  : ReturnType<typeof createConfig$>;
+
+export const ConfigSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+  encryptionKey: schema.conditional(
```

Comment:
Yeah, this bothers me as well, Kibana also isn't consistent in plugin IDs: it's a mix of `snake_case` and `camelCase`, even for NP plugins :cry: How do you feel about keeping `encryptionKey` (to be consistent with `security.encryptionKey` and `reporting.encryptionKey`), but change plugin ID to `encryptedSavedObjects`?

Reply:
> How do you feel about keeping encryptionKey (to be consistent with security.encryptionKey and reporting.encryptionKey), but change plugin ID to encryptedSavedObjects?

That was actually going to be my original suggestion, but I changed my mind 😄
I'm struggling to find evidence of this, but I thought we were taking steps to move all settings (and therefore plugin ids) to `snake_case` in order to be more consistent with Elasticsearch.

`camelCase` is arguably easier for us on TS/JS though, because we have a linter rule that enforces `camelCase` variable names, so any time you want to use a plugin, you'd have to do that conversion.

So in summary.... 🤷‍♂. If you don't want to make this change now, then maybe we can leave it as-is and try to have a broader discussion around naming?